Sources/DistributedActors/Cluster/ClusterShell.swift

```diff
@@ -396,8 +412,13 @@ extension ClusterShell {
             let uniqueBindAddress = self.selfNode
 
             // 1) failure detector:
-            let swimBehavior = SWIMActorShell.behavior(settings: self.settings.swim, clusterRef: context.myself)
-            self._swimRef = try context._spawn(SWIMActorShell.naming, props: SWIMActorShell.props, swimBehavior)
+//            let swimBehavior = SWIMActorShell.behavior(settings: self.settings.swim, clusterRef: context.myself)
+//            self._swimRef = try context._spawn(SWIMActorShell.naming, props: SWIMActorShell.props, swimBehavior)
+            Task {
+                await _Props.$forSpawn.withValue(SWIMActorShell.props) {
+                    self._swim = await SWIMActorShell(settings: self.settings.swim, clusterRef: context.myself, system: context.system)
+                }
+            }
```

Comment:
Can we try to make the cluster shell init async as well, this way we don't have to do the lock dance?

If definite initialization wil be preventing us we could get away by marking the `swim` var as `var swim: SWIM.Shell!` so we would be allowed to `await` and THEN initialize it, I think then?

Reply:
How about https://github.com/apple/swift-distributed-actors/pull/977/commits/d862fb20b5112b60411b02205115e5ad36ea6ab8?